integrationTests/relayers/ethToMultiversXWithChainSimulator_test.go

```diff
@@ -157,178 +162,467 @@ type keysHolder struct {
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexFalse,
+			mvxHexIsNative:   hexTrue,
+			ethIsMintBurn:    false,
+			ethIsNative:      true,
+		},
+		"error = invalid setup isNativeOnEthereum = true, isNativeOnMultiversX = true",
+		batchProcessor.ToMultiversX,
+	))
+	t.Run("ETH->MVX, ethNative = true, ethMintBurn = false, mvxNative = true, mvxMintBurn = true", testRelayersShouldNotExecuteTransfers(
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexTrue,
+			mvxHexIsNative:   hexTrue,
+			ethIsMintBurn:    false,
+			ethIsNative:      true,
+		},
+		"error = invalid setup isNativeOnEthereum = true, isNativeOnMultiversX = true",
+		batchProcessor.ToMultiversX,
+	))
+	t.Run("ETH->MVX, ethNative = true, ethMintBurn = true, mvxNative = true, mvxMintBurn = false", testEthContractsShouldError(
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexFalse,
+			mvxHexIsNative:   hexTrue,
+			ethIsMintBurn:    true,
+			ethIsNative:      true,
+		},
+	))
+	t.Run("ETH->MVX, ethNative = true, ethMintBurn = true, mvxNative = true, mvxMintBurn = true", testEthContractsShouldError(
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexTrue,
+			mvxHexIsNative:   hexTrue,
+			ethIsMintBurn:    true,
+			ethIsNative:      true,
+		},
+	))
+	t.Run("ETH->MVX, ethNative = false, ethMintBurn = true, mvxNative = false, mvxMintBurn = true", testEthContractsShouldError(
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexTrue,
+			mvxHexIsNative:   hexFalse,
+			ethIsMintBurn:    true,
+			ethIsNative:      false,
+		},
+	))
+	t.Run("MVX->ETH, ethNative = true, ethMintBurn = false, mvxNative = true, mvxMintBurn = false", testRelayersShouldNotExecuteTransfers(
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexFalse,
+			mvxHexIsNative:   hexTrue,
+			ethIsMintBurn:    false,
+			ethIsNative:      true,
+		},
+		"error = invalid setup isNativeOnEthereum = true, isNativeOnMultiversX = true",
+		batchProcessor.FromMultiversX,
+	))
 }
 
-func testTransfersBothWaysWithChainSimulatorAndConfig(t *testing.T, cfg testConfig) {
+func testRelayersShouldNotExecuteTransfers(
+	argsSimulatedSetup argSimulatedSetup,
+	expectedStringInLogs string,
+	direction batchProcessor.Direction,
+) func(t *testing.T) {
+	return func(t *testing.T) {
+		defer func() {
+			r := recover()
+			if r != nil {
+				require.Fail(t, "should have not panicked")
+			}
+		}()
+
+		argsSimulatedSetup.t = t
+		testSetup := prepareSimulatedSetup(argsSimulatedSetup)
+		defer testSetup.close(t)
+
+		checkESDTBalance(t, testSetup.testContext, testSetup.mvxChainSimulator, testSetup.mvxReceiverAddress, testSetup.mvxUniversalToken, "0", true)
+
+		createBatch(t, testSetup, direction)
 
-	defer func() {
-		r := recover()
-		if r != nil {
-			require.Fail(t, "should have not panicked")
+		// start a mocked log observer that is looking for a specific relayer error
+		chanCnt := 0
+		mockLogObserver := mock.NewMockLogObserver(expectedStringInLogs)
+		err := logger.AddLogObserver(mockLogObserver, &logger.PlainFormatter{})
```

Comment:
Interesting approach. However, care should be taken when using static methods & variables as used in the logger package.
Maybe add some comment warnings to not run tests in parallel from the current package? 
Better, create a dedicated package for these slow tests?

Reply:
moved to relayers_test

Reply:
I was referring to a new directory